services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/feedback_revision_trial.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: evidence_research_gen_ai_feedback_revision_trials
+#
+#  id                   :bigint           not null, primary key
+#  limit                :integer
+#  prompt               :text             not null
+#  results              :jsonb
+#  status               :string           not null
+#  temperature          :float            default(0.0), not null
+#  trial_duration       :float
+#  trial_errors         :text             default([]), not null, is an Array
+#  created_at           :datetime         not null
+#  updated_at           :datetime         not null
+#  feedback_revision_id :bigint           not null
+#  llm_id               :bigint           not null
+#
+module Evidence
+  module Research
+    module GenAI
+      class FeedbackRevisionTrial < ApplicationRecord
+        include HasStatus
+
+        belongs_to :feedback_revision
+        belongs_to :llm
+
+        has_many :feedback_revision_trial_llm_examples, dependent: :destroy
+        has_many :feedback_evaluations, through: :feedback_revision_trial_llm_examples
+
+        validates :prompt, presence: true
+        validates :temperature, presence: true
+
+        store_accessor :results, :accuracy
+
+        def run = FeedbackRevisionTrialRunner.run(self)
+
+        def update_results!(updates)
```

Comment:
Just noting that this has a similar race condition to [this](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb#L113-L125). If multiple jobs will write to this, you may need to update this to that form.

Reply:
Great point.  I forgot about this race condition.